code/modules/status_system/statusEffects.dm

```diff
@@ -3706,3 +3706,59 @@
 		message_admins("[key_name(src.owner)] regained their sanity and is no longer broken.")
 		logTheThing(LOG_ADMIN, src.owner, "regained their sanity and is no longer broken.")
 		mob_owner.mind.remove_antagonist(ROLE_BROKEN, ANTAGONIST_REMOVAL_SOURCE_EXPIRED)
+
+/datum/statusEffect/mimicDisguise
+	id = "mimic_disguise"
+	name = "Disguised"
+	desc = "Your speed and health are matching with your disguise."
+	icon_state = "mimicface"
+	visible = TRUE
+	movement_modifier = /datum/movement_modifier/mimic
+	var/last_speed = null
+	var/last_amount = null
+	var/speed_string = null
+
+	getTooltip()
+		var/mob/living/critter/mimic/mob_owner = src.owner
+		return "Health: [mob_owner.max_health], Speed: [speed_string]"
+
+	onUpdate()
+		..()
+		var/mob/living/critter/mimic/mob_owner = src.owner
+		var/new_amount = mob_owner.pixel_amount
+		if (new_amount != src.last_amount)
+			src.last_amount = new_amount
+			scale()
```

Comment:
Why not just call `setStatus` with the pixel amount instead of constantly checking it here?